dio/lib/src/options.dart

```diff
@@ -169,62 +172,62 @@ class Options extends _RequestConfig {
 
   /// Create a Option from current instance with merging attributes.
   Options merge({
-    String method,
-    int sendTimeout,
-    int receiveTimeout,
-    Map<String, dynamic> extra,
-    Map<String, dynamic> headers,
-    ResponseType responseType,
-    String contentType,
-    ValidateStatus validateStatus,
-    bool receiveDataWhenStatusError,
-    bool followRedirects,
-    int maxRedirects,
-    RequestEncoder requestEncoder,
-    ResponseDecoder responseDecoder,
+    String? method,
+    int? sendTimeout,
+    int? receiveTimeout,
+    Map<String, dynamic>? extra,
+    Map<String, dynamic>? headers,
+    ResponseType? responseType,
+    String? contentType,
+    ValidateStatus? validateStatus,
+    bool? receiveDataWhenStatusError,
+    bool? followRedirects,
+    int? maxRedirects,
+    RequestEncoder? requestEncoder,
+    ResponseDecoder? responseDecoder,
   }) {
     return Options(
       method: method ?? this.method,
       sendTimeout: sendTimeout ?? this.sendTimeout,
       receiveTimeout: receiveTimeout ?? this.receiveTimeout,
-      extra: extra ?? Map.from(this.extra ?? {}),
-      headers: headers ?? Map.from(this.headers ?? {}),
+      extra: extra ?? this.extra,
+      headers: headers ?? this.headers,
       responseType: responseType ?? this.responseType,
       contentType: contentType ?? this.contentType,
       validateStatus: validateStatus ?? this.validateStatus,
       receiveDataWhenStatusError:
           receiveDataWhenStatusError ?? this.receiveDataWhenStatusError,
       followRedirects: followRedirects ?? this.followRedirects,
       maxRedirects: maxRedirects ?? this.maxRedirects,
-      requestEncoder: requestEncoder,
+      requestEncoder: requestEncoder ?? this.requestEncoder,
       responseDecoder: responseDecoder ?? this.responseDecoder,
     );
   }
 }
 
 class RequestOptions extends Options {
   RequestOptions({
-    String method,
-    int sendTimeout,
-    int receiveTimeout,
+    String? method,
+    int? sendTimeout,
+    int? receiveTimeout,
     this.connectTimeout,
     this.data,
-    this.path,
-    this.queryParameters,
-    this.baseUrl,
+    String? path,
```

Comment:
Same as with the `BaseOptions` constructor. 
```suggestion
    this.path = '',
```